content/storage-history/history.textile

```diff
@@ -0,0 +1,633 @@
+    }
+    ```
+
+This gets a paginated set of historical presence messages for the channel. If the channel is configured to persist messages to disk, then message history will typically be available for 24 or 72 hours. If not, messages are only retained in memory by the Ably service for two minutes. See "persistence enabled":/storage-history/storage.
+
+On success, @resultPage@ contains a @PaginatedResult@ encapsulating an array of @PresenceMessage@ objects corresponding to the current page of results. @PaginatedResult@ supports pagination using @next()@ and @first()@ methods.
+
+On failure to retrieve presence event history, @err@ contains an @ErrorInfo@ object with the failure reason.
+
+h4(#presence-parameters). Presence parameters
+
+There is only one parameter for @history@: 
+
+- options := ensures that clients disconnected abruptly from Ably can resume their connection
+
+h4(#presence-option-parameters). Presence parameters
+
+These are the parameters you can add to your @options@ object, note that @untilAttach@ is only available for realtime:
+
+|_. Parameter |_. Description |
+| start | beginning of time earliest time in milliseconds since the epoch for any messages retrieved |
+| end | current time latest time in milliseconds since the epoch for any messages retrieved |
+| direction | backwards forwards or backwards |
+| limit | 100 maximum number of messages to retrieve up to 1,000 |
+
+h2(#historical-order). Ordering of historical messages
+
+The order in which historical messages are returned with History is based on the message timestamp that was assigned by the channel in the region that the message was published in. This ordering is what Ably calls the canonical global order.
+
+It is important to note that this is not necessarily the order that messages were received by a realtime client. The order in which each realtime client receives a message depends on which region the client is in. 
+
+Ably preserves ordering for a specific publisher on a specific channel but, for example, if two publishers in regions A and B publish "message 1" and "message 2" simultaneously, then it is very possible that a subscriber in region A will receive "message 1" before "message 2", but that a subscriber in region B will receive "message 2" before "message 1".
+
+There are instances where messages will not be in canonical global order:
+
+* Recent messages (less than two minutes old) are retrieved from live ephemeral storage and are still ordered by region. They only appear in the canonical global order if you enable "message persistence":/storage-history/storage, which also prevents duplication and missing messages.
+* You choose to retrieve historical messages only up to the point at which a client attaches to a channel. You would typically use this approach to bring a subscriber up to date as part of connection state recovery.
+
+h2(#continuous-history). Continuous history
```

Comment:
Should this section be under "retrieve channel history"?

Reply:
I've left this where it is (in the historical ordering section) as they are exceptions to the canonical global order as opposed to retrieving messages themselves.

Reply:
Sorry, was referring to the "continuous history" section rather than the bullets 🙂